.NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs

```diff
@@ -128,8 +128,10 @@ public static class DateTimeDefinitions
       public static readonly string TimeRegex2 = $@"(\b{TimePrefix}\s+)?(t)?{BaseDateTime.HourRegex}(\s*)?:(\s*)?{BaseDateTime.MinuteRegex}((\s*)?:(\s*)?{BaseDateTime.SecondRegex})?((\s*{DescRegex})|\b)";
       public static readonly string TimeRegex3 = $@"\b{BaseDateTime.HourRegex}\.{BaseDateTime.MinuteRegex}(\s*{DescRegex})(\s+{TimePrefix})?";
       public static readonly string TimeRegex4 = $@"\b{BasicTime}(\s*{DescRegex})?(\s+{TimePrefix})?\s+{TimeSuffix}\b";
+      public static readonly string TimeRegex4CS = $@"(?<=(?<DurationPrep>(pour|durée de)\s+)|){TimeRegex4}(?(DurationPrep)(?<!heures?))";
```

Comment:
Would it be possible to solve without conditional statements? These are not supported in regexes in some of the other target platforms, right?
Also, couldn't this be solved by some ambiguity filter at the time level to discard extracted times if the prefix is present?

Reply:
Duplicating regexes (one for .NET one for the rest) also introduces extra maintainability issues.

Reply:
solving it without conditional statement at the regex level seems much more complicate and the use of lookbehinds (which are also not supported everywhere) seems unavoidable. 
An ambiguity filter in the TimeExtractor seems a more viable solution, I will try it (I have discarded this approach because it means modifying the base code for a language specific problem which is also not recommended).

Reply:
There is a generic mechanism for ambiguity filters already and it's usually applied "at the end" of a type.
The filter dictionary itself stays in the language specific resource file.

Reply:
Which platform doesn't support lookbehinds?

Reply:
python supports only fixed width lookaround, and also java and javascript have some limitations (I encountered problems with lookarouds of infinite width or with multiple quantifiers).

Reply:
added method FilterAmbiguity to BaseTimeExtractor and removed conditional regexes.